app/services/subscriptions/override_service.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module Subscriptions
+  class OverrideService < BaseService
+    include ChargeModelAttributesHandler
+
+    def call(**args)
+      ActiveRecord::Base.transaction do
+        plan_params = handle_plan_args(args)
+
+        plan_result = ::Plans::CreateService.new.create(**plan_params)
+
+        return failed_plan_creation_result(plan_result) unless plan_result.success?
+
+        subscription_params = handle_subscription_args(args, plan_result.plan.id)
+        subscription_result = ::Subscriptions::CreateService.new.create(**subscription_params)
+
+        return failed_subscription_creation_result(subscription_result) unless subscription_result.success?
+
+        result.subscription = subscription_result.subscription
+      end
+
+      result
+    end
+
+    private
+
+    def handle_plan_args(args)
+      params = args[:plan]
+      params[:code] = "#{params[:code]}-#{SecureRandom.uuid}"
+
+      prepare_arguments(**params)
+        .merge(overridden_plan_id: args[:overridden_plan_id])
+        .merge(organization_id: args[:organization_id])
+    end
+
+    def handle_subscription_args(args, plan_id)
+      args.delete(:plan)
+      args.delete(:overridden_plan_id)
+      args[:plan_id] = plan_id
+
+      args
+    end
+
+    def failed_plan_creation_result(plan_result)
+      if plan_result.error_code == 'not_found'
+        result.fail!(
+          code: 'not_found',
+          message: 'Some resources have not been found while overriding plan',
+          details: plan_result.error_details,
+        )
+      else
+        result.fail!(
+          code: 'unprocessable_entity',
+          message: 'Validation error happened while overriding plan',
+          details: plan_result.error_details,
+        )
+      end
+    end
+
+    def failed_subscription_creation_result(subscription_result)
+      if subscription_result.error_code == 'missing_argument'
+        result.fail!(
+          code: 'not_found',
+          message: 'Some resources have not been found while creating subscription',
```

Comment:
Front end does not have real need for "human readable messages" as it uses error code for translation.

I'm currently working on an improvement of the error management, but in the meantime, I suggest to just return the sub-service error.

Reply:
agree, why we do not just return sub service result?!

Reply:
We mostly use `not_found` if any record could not be found (even though it is e.g. some other record's argument) so that's why I wrapped errors here (to avoid `missing_argument`) error. Also, wanted to add clarification in which part (plan or subscription) error happened. I was aware that FE is using just `key` but I was thinking that API could also use it (later realised I will need new service for API) :) 
It's changed now and I'm just passing errors if it will be refactored in Vincent's change anyways :)